docs/source/guides/feature_descriptions.rst

```diff
@@ -0,0 +1,107 @@
+                          agg_primitives=["mean", "sum", "mode", "n_most_common"],
+                          trans_primitives=["month", "hour"],
+                          max_depth=2,
+                          features_only=True)
+    features = {feature.get_name(): feature for feature in feature_defs}
+    list(features.values())
+
+By default, ``describe_feature`` uses the existing variable and entity names and the default primitive description templates to generate feature descriptions. 
+
+.. ipython:: python
+
+    ft.describe_feature(features['HOUR(date_of_birth)'])
+    ft.describe_feature(features['MEAN(sessions.SUM(transactions.amount))'])
+Improving Descriptions
+~~~~~~~~~~~~~~~~~~~~~~~
+While the default descriptions can be helpful, they can also be further improved by providing custom definitions of variables and features, and by providing alternative templates for primitive descriptions. 
+
+Feature Descriptions
+---------------------
+Custom feature definitions will get used in the description in place of the automatically generated description. This can be used to either better explain what a variable is, or to provide descriptions that take advantage of a user's existing knowledge about the data or domain. 
+
+.. ipython:: python
+
+    feature_descriptions = {
+        'customers: join_date': 'the date the customer joined',
+        'sessions: SUM(transactions.amount)': 'the total transaction amount for a session'}
+
+    ft.describe_feature(features['HOUR(join_date)'],
+                        feature_descriptions=feature_descriptions)
+
+    ft.describe_feature(features['MEAN(sessions.SUM(transactions.amount))'],
+                        feature_descriptions=feature_descriptions)
+
+
+The above replaces the variable name ``"join_date"`` with a more descriptive definition of what that variable represents in the dataset. In addition, it also provides a custom description of the intermediate feature ``SUM(transactions.amount)``. The description for ``MEAN(sessions.SUM(transactions.amount))``, which is built on top of ``SUM(transactions.amount)``, uses the custom description in place of the automatically generated one. Feature descriptions can be passed in as a dictionary that maps the custom descriptions to either the feature object itself or the unique feature name in the form ``"[entity_name]: [feature_name]"``, as shown above.
+
+Primitive Templates
+--------------------
+Primitives descriptions are generated using primitive templates. By default, these are defined using the ``description_template`` attribute on the primitive. Primitives without a template default to using the primitive name instead. Primitive description templates are string templates that take input feature descriptions as the positional arguments. These can be overwritten by mapping primitive instances or primitive names to custom templates and passing them into ``describe_feature`` through the ``primitive_templates`` argument. 
```

Comment:
> Primitives without a template default to using the primitive name instead.

This is referring to primitive.name as opposed to the primitIve's class name.  Do you think clarifying that would be necessary?

Reply:
Actually it should probably have a fallback if `primitive.name` isn't defined. We could clarify that it uses `primitive.name` if it's defined, otherwise it just uses the class name.